repos/backend/endpoints/files.js

```diff
@@ -43,7 +45,98 @@ const deleteFile = (app, config) => async (req, res) => {
   }
 }
 
+/**
+ * Recursively checks to find the parent node for a given item
+ * if a parent exists, it will add it as a child
+ * @param {Array<Object>} nodes 
+ * @param {string} parentPath 
+ * @param {Object} newItem
+ * 
+ * @returns {Boolean} - whether a parent node exists and push was successful
+ */
+const parentNodeExists = (nodes, parentPath, newItem) => {
+  const found = nodes.find((node) => {
+    return node.fullPath === parentPath
+      ? Boolean(node.children.push(newItem))
+      : node.children.length && parentNodeExists(node.children, parentPath, newItem)
+  })
+
+  return Boolean(found)
+}
+
+/**
+ * Gets the metadata of a path from the local filesystem
+ * @param {string} path - full path to the folder or file i.e '/keg/tap/tests/bdd/features'
+ * 
+ * @returns {Object} - Meta data containing {name, parent, type ( folder || file )} properties
+ */
+const getPathMeta = path => {
+  const pathSplit = path.split('/')
+  return {
+    name: pathSplit.pop(),
+    parent: pathSplit.join('/'),
+    type: fs.lstatSync(path).isDirectory() ? 'folder' : 'file',
+  }
```

Comment:
I think a simpler way to get the name and parent values would be to use the `path` module:

```js
const getPathMeta = filePath => {
  return {
    name: path.basename(filePath),
    parent: path.dirname(path),
    ...
  }
}
```

Reply:
Damn son, putting me to shame. Works for me

Reply:
nice. updated